### PR TITLE
Destination Audit logs

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Configures Kafka Clusters to send audit logs to centralized destination cluster. Must be deployed with Centralized RBAC enabled Architecture

** Does NOT Create Necessary RoleBindings **
Because destination cluster gets created after MDS and potentially other Managed Kafka Clusters, it is impossible to know the Audit Logs Destination Kafka Cluster ID, and thus impossible to create the necessary role bindings. We must document these post installation steps, which can be later automated if necessary:

```
# Get Principals used in Destination Cluster Connection
ansible -i ~/.cache/molecule/confluent.test/rbac-mds-mtls-custom-rhel/inventory kafka_broker -m import_role -a "name=confluent.kafka_broker tasks_from=set_principal.yml" -e listener="{{audit_logs_destination_listener}}"

kafka-broker2 | SUCCESS => {
    "kafka_broker_principal": "User:CN=kafka_broker,OU=QE IT,O=CONFLUENT,L=PaloAlto,ST=Ca,C=US"
}

# Get Destination Cluster Cluster Id
ssh <destination-kafka-node>
cluster_id=$(grep "cluster.id" /var/lib/kafka/data/meta.properties | cut -d '=' -f2)

# Create RoleBindings
confluent login <mds-kafka>:8090
<login as mds super user>

# using cluster registry which is recommended in our docs
confluent iam rolebinding create --principal User:kafka --role ResourceOwner --resource Topic:confluent-audit-log-events --prefix --cluster-name audit_logs

confluent iam rolebinding create --principal User:kafka  --role DeveloperWrite --resource Topic:confluent-audit-log-events --prefix --cluster-name audit_logs


confluent iam rolebinding create --principal User:<audit-log-admin> --role ResourceOwner --resource Topic:confluent-audit-log-events --prefix --kafka-cluster-id $cluster_id

confluent iam rolebinding create --principal User:<audit-log-writer> --role DeveloperWrite --resource Topic:confluent-audit-log-events --prefix --kafka-cluster-id $cluster_id
```

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Updated two multi rbac scenarios to test centralized audit logs

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.kafka_broker/tasks/audit_logs.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    status_code: 200
+  register: destination_cluster_id
+
+- debug:
+    msg: "{{destination_cluster_id.json.id}}"
+
+# confluent iam rolebinding create --principal User:<audit-log-admin> --role ResourceOwner --resource Topic:confluent-audit-log-events --prefix --kafka-cluster-id <kafka-destination-cluster-id>
+- name: Grant Audit Logs Principal ResourceOwner on confluent-audit-log-events Prefixed Topics
+  uri:
+    url: "{{mds_bootstrap_server_urls.split(',')[0]}}/security/1.0/principals/User:{{ sasl_plain_users.admin.principal }}/roles/ResourceOwner/bindings"
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/audit_logs.yml:15]: principal should not be hardcoded to the sasl plain admin

File: roles/confluent.kafka_broker/tasks/audit_logs.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/audit_logs.yml:1]: delete this file!

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 3]
```diff
@@ -243,6 +243,12 @@
     config_path: "{{ kafka_broker.config_file }}"
   when: kafka_broker_secrets_protection_enabled|bool
 
+# - name: Create Audit Logs Rolebindings
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml:246]: remove commented out code

File: roles/confluent.test/molecule/rbac-mds-kerberos-mtls-custom-rhel/verify.yml [hunk 4]
```diff
@@ -1,4 +1,88 @@
 ---
+- name: Verify Audit Logs Pt1
+  hosts: kafka_broker
+  gather_facts: false
+  tasks:
+    # MDS -> Audit logs destination cluster with non super-user Principal, needs rolebindings
+
+    # TODO this url needs to be a variable... how should it get set in inventory
```
Comment (domenicbove) [inline roles/confluent.test/molecule/rbac-mds-kerberos-mtls-custom-rhel/verify.yml:8]: remove this comment

File: roles/confluent.variables/vars/main.yml [hunk 5]
```diff
@@ -344,6 +344,32 @@ kafka_broker_properties:
     enabled: "{{kafka_broker_telemetry_ansible_labels_enabled}}"
     properties:
       confluent.telemetry.labels.confluent.ansible.playbooks.version: "{{confluent_ansible_branch}}"
+  audit_logs_destination:
+    enabled: "{{audit_logs_destination_enabled and rbac_enabled}}"
+    properties:
+      confluent.security.event.logger.exporter.kafka.bootstrap.servers: "{{audit_logs_destination_bootstrap_servers}}"
+      # TODO recommendation is to set this to false on origin clusters
+      confluent.security.event.logger.exporter.kafka.topic.create: 'true'
+      # TODO this is buggy here
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:353]: add verify tasks on this property

**Review by JumaX — APPROVED** [review]
LGTM.

**Pull request merged**